database-plugins/src/main/java/io/cdap/plugin/db/connector/DBConnector.java

```diff
@@ -71,69 +64,25 @@
 @Name(DBConnector.NAME)
 @Description("This connector creates connections to a Database.")
 @Category("Database")
-public class DBConnector implements DirectConnector {
+public class DBConnector extends AbstractDBConnector<DBConnectorConfig> implements DirectConnector {
   public static final String NAME = "Database";
-  public static final String ENTITY_TYPE_DATABASE = "DATABASE";
-  public static final String ENTITY_TYPE_SCHEMA = "SCHEMA";
-  public static final String[] SUPPORTED_TYPES = new String[]{"TABLE", "VIEW", "MATERIALIZED VIEW"};
 
-  private static final String RESULTSET_COLUMN_TABLE_NAME = "TABLE_NAME";
-  private static final String RESULTSET_COLUMN_TABLE_TYPE = "TABLE_TYPE";
-  private static final String RESULTSET_COLUMN_TABLE_SCHEM = "TABLE_SCHEM";
-  private static final String RESULTSET_COLUMN_DATA_TYPE = "DATA_TYPE";
-  private static final String RESULTSET_COLUMN_TYPE_NAME = "TYPE_NAME";
-  private static final String RESULTSET_COLUMN_DECIMAL_DIGITS = "DECIMAL_DIGITS";
-  private static final String RESULTSET_COLUMN_COLUMN_SIZE = "COLUMN_SIZE";
-  private static final String RESULTSET_COLUMN_COLUMN_NAME = "COLUMN_NAME";
-  private static final String RESULTSET_COLUMN_IS_NULLABLE = "IS_NULLABLE";
-
-  private DBConnectorConfig config;
-  private DriverCleanup driverCleanup;
+  private final DBConnectorConfig config;
 
   DBConnector(DBConnectorConfig config) {
+    super(config);
     this.config = config;
```

Comment:
can we have `config` in parent class be protected so that we don't need to keep to references to `config` in both parent class and this class ?

Reply:
We normally prefer private field in all classes, the parent class here doesn't actually know what the exact type the config is, it only knows it is a `DatasetProperties`, so good to have the config in each class to be cleaner

Reply:
hmm.. don't quite understand the point here. You are right, parent class just need to know it's a subclass of DatasetProperties. But subclass know its exact class when `extends AbstractDBConnector<DBConnectorConfig>`
parent class can only call methods of DatasetProperties while subclass can call methods of DBConnectorConfig, it doesn't prevent removing the redundancy here.